src/legacy/core_plugins/tile_map/public/components/wms_options.tsx

```diff
@@ -75,10 +80,18 @@ function WmsOptions({
 
         if (!wms.selectedTmsLayer && newBaseLayers.length) {
           setWmsOption('selectedTmsLayer', newBaseLayers[0]);
```

Comment:
setting this here will require:
- clicking on the "play" button
- or calling forceUpdateVis  (to send changes down to vis)

however i am wondering, should this happen here ? i see a few options:
- have tms stuff preloaded (at the vis registration time) ... then we should be able to set visualization defaults to the first layer.
- move the logic into map visualization .... if layer is not set use the first layer

i would prefer the first one.

Reply:
The [code](https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/region_map/public/plugin.ts#L74) which registers vis is run every time when the app opens (or the page is refreshed) no matter what plugin opened. I think the layers should be fetched only when map visualization opens. For example during redirecting to the editor but before rendering the default editor we can load tms layers and set the first layer. I tried to implement this idea. @ppisljar could you please have a look?